lib/api.js

```diff
@@ -634,16 +635,15 @@ module.exports = function(self, options) {
       });
     }
       
-    // Some were fetched fully, others are just join projections of a doc.
-    // Get the full thing, and also the live version, so we can compare,
-    // and build a new array of the full draft docs, with `_modified`
-    // properties added where appropriate.
-    //
-    // Also, a join result might be an object with `item` and `relationship`
-    // properties. Flatten that out so we just have an array of docs.
-
     function diff(callback) {
-      return async.eachSeries(related, function(doc, callback) {
+      var ids = _.map(related, function(item) {
+        return item._id || (item.item && item.item._id);
+      });
+      ids = _.uniq(ids);
+      ids = _.filter(ids, function(id) {
+        return !!id;
+      });
```

Comment:
Seems like a opportunity to chain these methods?

```
var ids = _.map(related, function(item) {
    return item._id || (item.item && item.item._id);
  })
  .uniq(ids)
  .filter(ids, function(id) {
    return !!id;
  });
```

Reply:
Actually, if we're going to do that, then we have to use _.chain(related) to start the chain and _.value() at the end of the chain so lodash knows where to stop. Or we can use _(related) as a shorthand. I think the existing code is clear and correct, so I'm going to leave it in place for now, but we should look at using lodash chaining more in future.